### PR TITLE
Remove instances of random.sample from sets (deprecated in Python 3.9)

### LINKED ISSUE #4555: Random sampling from set has been deprecated in Python 3.9
Python 3.9 has deprecated calling `random.sample` on sets, so users will now see a warning if they do something like:

```python
>>> import networkx as nx
>>> G = nx.path_graph(25)
>>> import random
>>> random.sample(G.nodes(), k=3)
DeprecationWarning: Sampling from a set deprecated
since Python 3.9 and will be removed in a subsequent version.
[15, 1, 2]
```

It doesn't look like this will have an affect on `PythonRandomInterface`, but there are a few instances of the `sample(G.nodes(), **kwargs)` in the test suite which should be fixed.

### PR DESCRIPTION
**Pull request by rossbar**
Closes #4555

I took a conservative approach and used `sorted` to convert sets to lists. This may cause a performance hit, but I figured this was safer from the perspective of reproducibility, though I'm not sure it's necessary. In some circumstances it may be sufficient to simply convert to `list` directly.

### CODE REVIEW

File: networkx/algorithms/centrality/betweenness.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def betweenness_centrality(
     if k is None:
         nodes = G
     else:
-        nodes = seed.sample(G.nodes(), k)
+        nodes = seed.sample(sorted(G.nodes()), k)
```
Comment (dschult) [inline networkx/algorithms/centrality/betweenness.py:123]: Unfortunately, nodes are not sortable in general.Using `sorted` should be fine in tests where we know what the nodes are, but in the package code itself we should use `list`.

Comment (rossbar) [reply networkx/algorithms/centrality/betweenness.py]: Yup this was an oversight on my part - should be fixed in 2e2a63e.

**Review by dschult — APPROVED** [review]
LGTM

**Review by MridulS — APPROVED** [review]

**Comment by dschult** [comment]
Yay!   Approval works!  Now does merging work?
We've got 2 approvals Mridul, want to merge this one?

**Pull request merged**